From 43476eafc3482f906e678e600334c710b3ee1d2d Mon Sep 17 00:00:00 2001
From: LuK1337 <priv.luk@gmail.com>
Date: Sun, 26 Aug 2018 20:47:08 +0200
Subject: [PATCH 18/19] soong_config: Allow whitelisted processes to use
 destroyed mutex

Change-Id: I41e5521122f554b60c724c845d19f2bec798dc3c

# Conflicts:
#	build/soong/android/variable.go
#	build/soong/soong_config.mk
---
 build/soong/android/variable.go | 4 ++++
 build/soong/soong_config.mk     | 1 +
 2 files changed, 5 insertions(+)

diff --git a/build/soong/android/variable.go b/build/soong/android/variable.go
index c4ed04d4..b9555c23 100644
--- a/build/soong/android/variable.go
+++ b/build/soong/android/variable.go
@@ -14,6 +14,9 @@ type Product_variables struct {
 	}
 	Target_exfat_driver struct {
 		Cflags []string
+        }
+	Target_destroyed_mutex_usage_whitelist struct {
+		Cppflags []string
 	}
 	Target_shim_libs struct {
 		Cppflags []string
@@ -41,6 +44,7 @@ type ProductVariables struct {
 	Needs_text_relocations  *bool `json:",omitempty"`
 	Specific_camera_parameter_library  *string `json:",omitempty"`
 	Target_exfat_driver  *string `json:",omitempty"`
+	Target_destroyed_mutex_usage_whitelist  *string `json:",omitempty"`
 	Target_shim_libs  *string `json:",omitempty"`
 	Uses_generic_camera_parameter_library  *bool `json:",omitempty"`
 	Uses_nvidia_enhancements  *bool `json:",omitempty"`
diff --git a/build/soong/soong_config.mk b/build/soong/soong_config.mk
index ef78a6fb..6f7d5643 100644
--- a/build/soong/soong_config.mk
+++ b/build/soong/soong_config.mk
@@ -11,6 +11,7 @@ $(call add_json_bool, Needs_legacy_camera_hal1_dyn_native_handle, $(filter true,
 $(call add_json_bool, Needs_text_relocations, $(filter true,$(TARGET_NEEDS_PLATFORM_TEXT_RELOCATIONS)))
 $(call add_json_str, Specific_camera_parameter_library, $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY))
 $(call add_json_str,  Target_exfat_driver, $(TARGET_EXFAT_DRIVER))
+$(call add_json_str, Target_destroyed_mutex_usage_whitelist, $(TARGET_DESTROYED_MUTEX_USAGE_WHITELIST))
 $(call add_json_str, Target_shim_libs, $(TARGET_LD_SHIM_LIBS))
 $(call add_json_bool, Uses_generic_camera_parameter_library, $(if $(TARGET_SPECIFIC_CAMERA_PARAMETER_LIBRARY),,true))
 $(call add_json_bool, Uses_nvidia_enhancements,              $(filter TRUE,$(NV_ANDROID_FRAMEWORK_ENHANCEMENTS)))
-- 
2.17.1

